pgvector/runtime/src/main/java/io/quarkiverse/langchain4j/pgvector/PgVectorEmbeddingStore.java

```diff
@@ -0,0 +1,293 @@
+     * @param textSegment Original content that was embedded.
+     * @return The auto-generated ID associated with the added embedding.
+     */
+    @Override
+    public String add(Embedding embedding, TextSegment textSegment) {
+        String id = randomUUID();
+        addInternal(id, embedding, textSegment);
+        return id;
+    }
+
+    /**
+     * Adds multiple embeddings to the store.
+     *
+     * @param embeddings A list of embeddings to be added to the store.
+     * @return A list of auto-generated IDs associated with the added embeddings.
+     */
+    @Override
+    public List<String> addAll(List<Embedding> embeddings) {
+        List<String> ids = embeddings.stream().map(ignored -> randomUUID()).collect(toList());
+        addAllInternal(ids, embeddings, null);
+        return ids;
+    }
+
+    /**
+     * Adds multiple embeddings and their corresponding contents that have been embedded to the store.
+     *
+     * @param embeddings A list of embeddings to be added to the store.
+     * @param embedded A list of original contents that were embedded.
+     * @return A list of auto-generated IDs associated with the added embeddings.
+     */
+    @Override
+    public List<String> addAll(List<Embedding> embeddings, List<TextSegment> embedded) {
+        List<String> ids = embeddings.stream().map(ignored -> randomUUID()).collect(toList());
+        addAllInternal(ids, embeddings, embedded);
+        return ids;
+    }
+
+    /**
+     * Finds the most relevant (closest in space) embeddings to the provided reference embedding.
+     *
+     * @param referenceEmbedding The embedding used as a reference. Returned embeddings should be relevant (closest) to this
+     *        one.
+     * @param maxResults The maximum number of embeddings to be returned.
+     * @param minScore The minimum relevance score, ranging from 0 to 1 (inclusive).
+     *        Only embeddings with a score of this value or higher will be returned.
+     * @return A list of embedding matches.
+     *         Each embedding match includes a relevance score (derivative of cosine distance),
+     *         ranging from 0 (not relevant) to 1 (highly relevant).
+     */
+    @Override
+    public List<EmbeddingMatch<TextSegment>> findRelevant(Embedding referenceEmbedding, int maxResults, double minScore) {
+        List<EmbeddingMatch<TextSegment>> result = new ArrayList<>();
+        try (Connection connection = setupConnection()) {
+            String referenceVector = Arrays.toString(referenceEmbedding.vector());
+            String query = String.format(
+                    "WITH temp AS (SELECT (2 - (embedding <=> '%s')) / 2 AS score, embedding_id, embedding, text, metadata FROM %s) SELECT * FROM temp WHERE score >= %s ORDER BY score desc LIMIT %s;",
+                    referenceVector, table, minScore, maxResults);
+            PreparedStatement selectStmt = connection.prepareStatement(query);
```

Comment:
You have to close the statement, otherwise you get leaks. 
This shows in the log:
```
2023-11-29 11:06:13,666 WARN  [io.agr.pool] (executor-thread-1) Datasource '<default>': JDBC resources leaked: 1 ResultSet(s) and 2 Statement(s)
```

Multiple usages in this class are missing this

Reply:
Let me check this, and probably the upstream project would need to fix this as well (since I copy pasted :) )

Reply:
Ah yes, maybe a good opportunity to contribute to that project too :)